### PR TITLE
Fixes #1454

### LINKED ISSUE #1454: Feature request: #defines to test RDKit version
It would be great to have compile-time macros for version checking. A related project has macros that may be useful (https://github.com/openbabel/openbabel/blob/master/src/config.h.cmake).

I see that there are runtime checks I can do, and will use these as a workaround.

### PR DESCRIPTION
**Pull request by greglandrum**
I think this is a reasonable solution to the problem, but feedback from @baoilleach and @bp-kelley would be welcome.

### CODE REVIEW

File: Code/cmake/Modules/RDKitUtils.cmake [hunk 1]
```diff
@@ -9,6 +9,10 @@ set(RDKit_RELEASENAME "${RDKit_Year}.${RDKit_Month}")
 if (RDKit_Revision)
   set(RDKit_RELEASENAME "${RDKit_RELEASENAME}.${RDKit_Revision}")
   set(RDKit_VERSION "${RDKit_VERSION}.${RDKit_Revision}")
+  if (RDKit_RevisionModifier)
```
Comment (baoilleach) [inline Code/cmake/Modules/RDKitUtils.cmake]: Do you really need this "if statement"? Won't this always be defined, either as empty string or having a value? (Same goes for the outer "if statement" about RDKit_Revision)

**Comment by baoilleach** [comment]
Looks good to me. I've added a minor style comment.

File: Code/cmake/Modules/RDKitUtils.cmake (hunk 1 above)
Comment (greglandrum) [reply Code/cmake/Modules/RDKitUtils.cmake]: good point. that's been revised

**Comment by bp-kelley** [comment]
I think commit https://github.com/rdkit/rdkit/pull/1838/commits/3dce006009e47d5259b667a759440950db1ec2ea highlights my concern with hard-coding.  If git is available when running cmake, I would rather use our current versioning system that we use in anaconda.  I.e. tag + number of commits since the tag.

**Comment by greglandrum** [comment]
I think I agree with the idea, but since that's a larger change to the versioning scheme, it's maybe one for a separate pull request.
This would be something directly helpful for the next patch release, that would be more of a "next release" thing.

**Pull request merged**